Azure/azure-cli
# [Profile] Return subscriptions even with --allow-no-subscriptions

Fix #11059: `az login --allow-no-subscriptions` fails if there are subscriptions in the tenant

In `az login --allow-no-subscriptions`, only tenants with no subscriptions are listed and available subscriptions are missing from the subscription list. If the only tenant has some subscriptions, this command fails. 

This PR fixes this and honor the word "allow" so that available subscriptions are preserved and other tenants with no subscriptions are appended to the subscription list. 

To test:

```
az login --allow-no-subscriptions
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).
